include/i2c.h

```diff
@@ -175,6 +204,105 @@ static inline int _impl_i2c_transfer(struct device *dev,
 	return api->transfer(dev, msgs, num_msgs, addr);
 }
 
+/**
+ * @brief Attaches the provided funcs and address as Slave device
+ *
+ * This routine starts the I2C Controller in Slave mode and uses
+ * the provided funcs, address and priv to send events to the
+ * Slave driver implementation.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param address Address of the I2C Slave to match on the bus
+ * @param funcs The Slave API funcs used by the controller to send bus events
+ * @param priv Private data passed along the bus events
+ *
+ * @retval 0 Is successful
+ * @retval -EINVAL If parameters are invalid
+ * @retval -EIO General input / output error.
+ */
+__syscall int i2c_slave_attach(struct device *dev, u8_t address,
```

Comment:
We should use u16_t for the `address` parameter. Also, let's shorten `address` to `addr` to be consistent with existing I2C API.

Reply:
Yes, I must add the 10bit address support, and a flag is missing here somewhere, maybe in a dedicated struct containing the context, address and flags.

Reply:
Yes, actually we do need a flag to set the addressing mode.